modules/library/main/src/main/java/org/geotools/filter/function/FilterFunction_arrayAnyMatch.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2019, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.function;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import java.util.Arrays;
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.filter.capability.FunctionName;
+
+public class FilterFunction_arrayAnyMatch extends FunctionExpressionImpl {
+    public static FunctionName NAME =
+            new FunctionNameImpl(
+                    "arrayAnyMatch",
+                    parameter(
+                            "arrayAnyMatch",
+                            Boolean.class,
+                            "Any",
+                            "Returns true if any array value matches candidate."),
+                    parameter("array", Object[].class, "Array", "array of values"),
+                    parameter("candidate", Object.class, "Candidate", "value to match with array"));
+
+    public FilterFunction_arrayAnyMatch() {
+        super(NAME);
+    }
+
+    public Object evaluate(Object feature) {
+        Object[] array;
+        Object candidate;
+
+        try { // attempt to get array and perform conversion
+            array = (Object[]) getExpression(0).evaluate(feature);
```

Comment:
if you expect an `Object[]` then ask for it rather than eating exceptions:
```
array = getExpression(0).evalulate( feature, Object[] )
```

This approach reduce the amount if error checking boilerplate and allows folks to introduce conversions, for example allowing your function to handle an array represented as a string.